test/unit/spec/kbaseNarrativeSpec.js

```diff
@@ -97,51 +104,50 @@ define([
             jasmine.Ajax.uninstall();
             Jupyter.notebook = null;
             NarrativeLogin.clearTokenCheckTimers();
+            NarrativeLogin.destroy();
             Mocks.clearAuthToken();
+            // clear all jquery event listeners set up by either NarrativeLogin or anything else
+            $(document).off();
+            $([Jupyter.events]).off();
+            document.body.innerHTML = '';
         });
 
         it('Should instantiate', () => {
             const narr = new Narrative();
             expect(narr.maxNarrativeSize).toBe('10 MB');
+            narr.loadingWidget.clearTimeout();
```

Comment:
if you move creating the new narrative to the `beforeEach`, you can put `narr.loadingWidget.clearTimeout();` in the `afterEach`.

Reply:
Good idea. Fixed.